### PR TITLE
added jdk 8 instructions for macOS in SETUP.md

### PR DESCRIPTION
**Pull request by GameCenterJerry**
added necessary commands and download links

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -49,6 +49,27 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
 
 If you have errors with a package missing please make sure you have setup your environment, and are using Oracle JDK 8.
 
+To check which java you are using do 
+`java -version` in a command prompt or terminal.
+If you are using Oracle JDK 8+ it will not work because the Java distributions above JDK 8 using may not have the needed javax classes.
+
+Oracle JDK 8 download: https://www.oracle.com/java/technologies/javase/javase-jdk8-downloads.html
```
Comment (l1ving) [inline SETUP.md:56]: I'd rather link to OpenJDK please

**Review by l1ving — CHANGES_REQUESTED** [review]

File: SETUP.md [hunk 2]
```diff
@@ -49,6 +49,27 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
 
 If you have errors with a package missing please make sure you have setup your environment, and are using Oracle JDK 8.
 
+To check which java you are using do 
+`java -version` in a command prompt or terminal.
+If you are using Oracle JDK 8+ it will not work because the Java distributions above JDK 8 using may not have the needed javax classes.
```
Comment (l1ving) [inline SETUP.md:54]: OpenJDK 8*

File: SETUP.md (hunk 1 above)
Comment (GameCenterJerry) [reply SETUP.md]: Done

File: SETUP.md [hunk 3]
```diff
@@ -49,6 +49,27 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
...
+
+Open JDK 8 download: https://openjdk.java.net/install/
+#### macOS guide
+In order to get JDK 8, run the following command:
+`% /usr/libexec/java_home -V`
+If you see something like
+
+`% 1.8.0_202, x86_64:	"Java SE 8"	/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home`
```
Comment (l1ving) [inline SETUP.md:62]: This version shouldn't be hardcoded, just indicate they have to change the version. You'd be surprised how many people just blindly copy and paste commands

File: SETUP.md [hunk 4]
```diff
@@ -49,6 +49,27 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
...
+
+in the list then you've got JDK 8 installed. 
+In order to get JDK 8 running in the **current terminal window** you will have to run this command: 
+
+`% export JAVA_HOME=$(/usr/libexec/java_home -v 1.8)`
+
+You will have to compile the file in the same terminal window.
```
Comment (l1ving) [inline SETUP.md:69]: Suggest adding it to their PATH so they don't have to do it every time?

**Review by l1ving — CHANGES_REQUESTED** [review]

File: SETUP.md (hunk 3 above)
Comment (GameCenterJerry) [reply SETUP.md:62]: true. Changing path can be a little tricky though and I wanted to make it easier to find. But which line do you suggest i remove/edit?

File: SETUP.md (hunk 4 above)
Comment (GameCenterJerry) [reply SETUP.md:69]: Will do. I'll link a guide to changing PATH so I don't have to put a giant hunk of info in the SETUP.md

File: SETUP.md (hunk 2 above)
Comment (GameCenterJerry) [reply SETUP.md:54]: done

File: SETUP.md (hunk 3 above)
Comment (l1ving) [reply SETUP.md:62]:
```diff
- `% 1.8.0_202, x86_64:	"Java SE 8"	/Library/Java/JavaVirtualMachines/jdk1.8.0_202.jdk/Contents/Home`
+ `% 1.8.0_VERSION, x86_64:	"Java SE 8"	/Library/Java/JavaVirtualMachines/jdk1.8.0_VERSION.jdk/Contents/Home`
```

File: SETUP.md (hunk 4 above)
Comment (l1ving) [reply SETUP.md:69]: Eg just add that to your `.zshrc` at the end

File: SETUP.md (hunk 3 above)
Comment (GameCenterJerry) [reply SETUP.md]: changed it

File: SETUP.md (hunk 4 above)
Comment (GameCenterJerry) [reply SETUP.md]: Done (I think)

File: SETUP.md [hunk 5]
```diff
@@ -49,6 +49,28 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
...
+If you see something like
+
+`% 1.8.0_VERSION, x86_64:	"Java SE 8"	/Library/Java/JavaVirtualMachines/jdk1.8.0_VERSION.jdk/Contents/Home`
+
+in the list then you've got JDK 8 installed. 
+In order to get JDK 8 running in the **current terminal window** you will have to run this command: 
+
+`% export JAVA_HOME=$(/usr/libexec/java_home -v 1.8)`
+
+To add OpenJDK 8 to your PATH try this guide: http://www.sajeconsultants.com/how-to-set-java_home-on-mac-os-x/?utm_source=rss&utm_medium=rss&utm_campaign=how-to-set-java_home-on-mac-os-x. If you're using bash change the .bachrc and if you're using zsh change the .zshrc
```
Comment (l1ving) [inline SETUP.md:71]: Just change this to "Add the export line to the end of your `.zshrc` / `.bashrc` if you want it to apply to each new terminal" and you're good to go

**Review by l1ving — CHANGES_REQUESTED** [review]

File: SETUP.md (hunk 5 above)
Comment (GameCenterJerry) [reply SETUP.md]: It is done.

**Review by l1ving — APPROVED** [review]

File: SETUP.md [hunk 6]
```diff
@@ -49,6 +49,28 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
 
 If you have errors with a package missing please make sure you have setup your environment, and are using Oracle JDK 8.
 
+To check which java you are using do 
+`java -version` in a command prompt or terminal.
+If you are using OpenJDK 8+ it will not work because the Java distributions above JDK 8 using may not have the needed javax classes.
```
Comment (CDAGaming) [inline SETUP.md:54]: Shouldn't this be OpenJDK 9+, to resemble OpenJDK 9 and above not working. I read this as "If you are using OpenJDK 8 and above, which from my own experience, openjdk 8 works fine

File: SETUP.md [hunk 7]
```diff
@@ -49,6 +49,28 @@ On Mac OSX and Linux, use `./gradlew` instead of `gradlew`.
...
+
+`% 1.8.0_VERSION, x86_64:	"Java SE 8"	/Library/Java/JavaVirtualMachines/jdk1.8.0_VERSION.jdk/Contents/Home`
+
+in the list then you've got JDK 8 installed. 
+In order to get JDK 8 running in the **current terminal window** you will have to run this command: 
+
+`% export JAVA_HOME=$(/usr/libexec/java_home -v 1.8)`
+
+To add OpenJDK 8 to your PATH add the export line to the end of your `.zshrc / .bashrc` if you want it to apply to each new terminal. If you're using bash change the .bachrc and if you're using zsh change the .zshrc
+#### Windows 10 guide (currently working on it)
```
Comment (CDAGaming) [inline SETUP.md:72]: A windows 10 guide when you named your pr exclusive to macos would be declared out of scope, and should be in a separate pr if you indeed have plans of doing so.

**Review by CDAGaming — CHANGES_REQUESTED** [review]
Two minor requests, then LGTM

File: SETUP.md (hunk 6 above)
Comment (GameCenterJerry) [reply SETUP.md:54]: What I mean is "If you use anything above JDK 8 it MIGHT not work"

File: SETUP.md (hunk 7 above)
Comment (GameCenterJerry) [reply SETUP.md:72]: True, I'll remove it. If I'm gonna do a windows guide I'll do a seperate PR as you said.

**Comment by GameCenterJerry** [comment]
Added some minor edits as requested by @CDAGaming 

(Removed windows guide and made the JDK version more understandable.)

**Review by CDAGaming — APPROVED** [review]
LGTM!
